src/e2e-test/java/io/cdap/plugin/common/ValidationHelper.java

```diff
@@ -0,0 +1,278 @@
+import java.nio.charset.StandardCharsets;
+import java.time.LocalDate;
+import java.time.LocalDateTime;
+import java.time.LocalTime;
+import java.time.ZoneOffset;
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeParseException;
+import java.time.temporal.ChronoUnit;
+
+import java.util.Base64;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+import java.util.stream.StreamSupport;
+
+/**
+ * Validation Helper.
+ */
+public class ValidationHelper {
+  private static final String PROJECT_ID = PluginPropertyUtils.pluginProp("projectId");
+  private static final Gson gson = new Gson();
+
+  public static boolean validateBQDataToGCS(String table, String bucketName) throws IOException,
+    InterruptedException {
+    Map<String, JsonObject> expectedData = new HashMap<>();
+    Map<String, JsonObject> actualData = listBucketObjects(bucketName);
+
+    getBigQueryTableData(table, expectedData);
+
+    boolean isMatched = matchJsonMaps(expectedData, actualData);
+
+    if (isMatched) {
+      System.out.println("The data is matched.");
+    } else {
+      System.out.println("The data is not matched.");
+    }
+
+    return isMatched;
+  }
+
+  public static boolean validateGCSDataToBQ(String bucketName, String table) throws IOException,
+    InterruptedException {
+    Map<String, JsonObject> expectedData = new HashMap<>();
+    Map<String, JsonObject> actualData = listBucketObjects(bucketName);
+
+    getBigQueryTableData(table, expectedData);
+
+    boolean isMatched = matchJsonMaps(expectedData, actualData);
+
+    if (isMatched) {
+      System.out.println("The data is matched.");
+    } else {
+      System.out.println("The data is not matched.");
+    }
+
+    return isMatched;
+  }
+
+  public static Map<String, JsonObject> listBucketObjects(String bucketName) {
+    Map<String, JsonObject> dataMap = new HashMap<>();
+    Storage storage = StorageOptions.newBuilder().setProjectId(PROJECT_ID).build().getService();
+    Page<Blob> blobs = storage.list(bucketName);
+
+    List<Blob> bucketObjects = StreamSupport.stream(blobs.iterateAll().spliterator(), true)
+      .filter(blob -> blob.getSize() != 0)
+      .collect(Collectors.toList());
+
+    Stream<String> objectNamesWithData = bucketObjects.stream().map(blob -> blob.getName());
+    List<String> bucketObjectNames = objectNamesWithData.collect(Collectors.toList());
+
+    if (!bucketObjectNames.isEmpty()) {
+      String objectName = bucketObjectNames.get(0);
+      if (objectName.contains("part-r")) {
+        Map<String, JsonObject> dataMap2 = fetchObjectData(PROJECT_ID, bucketName, objectName);
+        dataMap.putAll(dataMap2);
+      }
+    }
+
+    return dataMap;
+  }
+
+  private static Map<String, JsonObject> fetchObjectData(String projectId, String bucketName, String objectName) {
+    Map<String, JsonObject> dataMap = new HashMap<>();
```

Comment:
Logic looks good try to use more relevant naming conventions.

Reply:
Done